artemis-server/src/main/java/org/apache/activemq/artemis/core/server/federation/address/FederatedAddress.java

```diff
@@ -113,7 +116,31 @@ public void start() {
     */
    @Override
    public synchronized void afterCreateQueue(Queue queue) {
-      createRemoteConsumer(queue);
+      conditionalCreateRemoteConsumer(queue);
+   }
+
+   private void conditionalCreateRemoteConsumer(Queue queue) {
+      //Always create demand if not a federated queue
+      if (!queue.getName().toString().startsWith(FEDERATED_QUEUE_PREFIX)) {
```

Comment:
@cshannon im still struggling to understand why this change is needed. It seems to be breaking some setups i have.

Reply:
@michaelandrepearce - This change is primarily there for the use case of 2 brokers each configured with a downstream with each other.  (so bi-directional).  If a consumer is created on broker0 then a federated consumer should be created on broker 1 and there should just be 1 consumer on each broker.  However, without this change you end up with a federated consumer on broker 0 as well (so 2 total consumers) eventhough there is no demand from a local consumer on broker 1.  I will push an update in a minute with a new test to show this.

Reply:
On federated address through it wasn't meant to be demand based on consumer, it was purely meant to be demand based on if there is a queue, e.g. multicast dispatch over the joined brokers. e.g. producers producers to "foo" on brokerA if brokerB had a queue under the address then a remote queue would be created on brokerA and the message would dispatch to brokerB, regardless if that queue had subscriber or not, e.g. handles cases of durable subscriptions, what was nice, is JMS non durable subs, as soon as consumer goes anyhow queue would go, and thus demand goes, as such demand just needed to be down to if queue or not.

Reply:
e.g. address level federation was desgined to be queue demand based, and queue level federation is designed to be consumer demand based